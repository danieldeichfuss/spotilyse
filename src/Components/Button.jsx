import React from "react";

export default function Button({ text, callback }) {
  return (
    <button className="Button" onClick={callback}>
      {text}
    </button>
  );
}
