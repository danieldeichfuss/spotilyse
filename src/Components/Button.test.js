import React from "react";
import ReactDOM from "react-dom";
import Button from "./Button";
import renderer from "react-test-renderer";

const text = "Button Text";
const callback = jest.fn();

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Button text={text} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders page correctly", () => {
  const component = renderer.create(<Button text={text} callback={callback} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
