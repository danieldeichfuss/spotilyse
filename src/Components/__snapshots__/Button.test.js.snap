// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders page correctly 1`] = `
<button
  className="Button"
  onClick={[Function]}
>
  Button Text
</button>
`;
